javascript/javascript3/week3/TASK3-Shopping cart/index.js

```diff
@@ -1,130 +1,149 @@
-const totalPrice = document.getElementById("total-price");
-
-class Currency {
-  constructor(name, rate) {
-    this.name = name;
-    this.rate = rate;
-  }
-}
-
-const USD = new Currency("USD", 7.5);
-const EUR = new Currency("EUR", 8.2);
-const HUF = new Currency("HUF", 46.2);
+(function () {
```

Comment:
Alright, there are a few things wrong with this approach: 
1. To understand the scope, you need to scroll a few pages
2. While this can technically work, but then have methods ... inside a class ... inside a function. 
3. At line 1 i don't know if this if a function _declaration_ or a function _invocation_, so i have to scroll until the end to find out
A better approach would be to to just have a _main()_ or an _async main()_ with only one global call at the end. The main() would be **outside** the class, not enclosing it.

Reply:
This exercise is small, but normally you would have a separate file for each class, with as few dependencies as possible.